### DISCUSSION TITLE
Create record with own id

### DISCUSSION THREAD

**Discussion by starredev** [pocketbase/pocketbase#2703] [Q&A] [answered]
In pocketbase I see it is possible to create records with creating your own unique identifier.
When I try to create a record in the js-sdk it will not store my own generated id.

Any others options?

**Comment by ganigeorgiev** [comment, marked as answer]
> When I try to create a record in the js-sdk it will not store my own generated id.

Is there any validation error that is returned? Note that the `id` currently has 15 chars min limit (_it may get removed in the future_).

**Reply by starredev** [reply]
> > When I try to create a record in the js-sdk it will not store my own generated id.
> 
> Is there any validation error that is returned? Note that the `id` currently has 15 chars min limit (_it may get removed in the future_).

Ah I see it must be exactly 15.
I hope in the near future there will be an limitation not less than {}.

Did you ever think about making relations not only based on record id?
Im working with jsonRpc with uuid messageId and it would be nice if I can create a relation between messageId in 2 collections

**Reply by ganigeorgiev** [reply]
> Did you ever think about making relations not only based on record id?

For now there are no plans for this as it may complicate too much the internals.
